reactfire/firebaseApp/sdk.tsx

```diff
@@ -1,28 +1,31 @@
 import { useFirebaseApp } from '..';
-import { preloadObservable, useObservable } from '../useObservable';
+import { useObservable } from '../useObservable';
 import { from } from 'rxjs';
-
-type RemoteConfig = import('firebase/app').remoteConfig.RemoteConfig;
-type Storage = import('firebase/app').storage.Storage;
-type Firestore = import('firebase/app').firestore.Firestore;
-type Performance = import('firebase/app').performance.Performance;
-type Messaging = import('firebase/app').messaging.Messaging;
-type Functions = import('firebase/app').functions.Functions;
-type Database = import('firebase/app').database.Database;
-type Auth = import('firebase/app').auth.Auth;
-
-type FirebaseSDK =
-  | (() => firebase.analytics.Analytics)
-  | (() => firebase.auth.Auth)
-  | ((url?: string) => firebase.database.Database)
-  | (() => firebase.firestore.Firestore)
-  | ((region?: string) => firebase.functions.Functions)
-  | (() => firebase.messaging.Messaging)
-  | (() => firebase.performance.Performance)
-  | (() => firebase.remoteConfig.RemoteConfig)
-  | ((url?: string) => firebase.storage.Storage);
-
-enum SDK {
+import { DEFAULT_APP_NAME } from '../index';
+
+type App = import('firebase/app').app.App;
+type RemoteConfig = Omit<typeof import('firebase/app').remoteConfig, "*"> & import('firebase/app').remoteConfig.RemoteConfig;
+type Storage = Omit<typeof import('firebase/app').storage, "*"> & import('firebase/app').storage.Storage;
```

Comment:
What does this `Omit` syntax do?

Reply:
It’s strange but this strips the constructor from the type, I’m going to clean this up more before pulling this in. I think I do want to go with the pattern that `useAuth == firebase.auth` and `useAuth() == firebase.auth()` to better mirror the JS SDK. Also chatting with @Feiyang1 to figure out why the props aren’t present on the namespace, so hopefully I don’t have to lean on Proxy & the internal API there.

Reply:
Unsolicited opinion warning but: to my mind, this makes a lot more sense - the closer the results returned from hooks like useFirestore() are to the vanilla firebase sdk the better, and it can take advantage of all the types.

Reply:
Addressed.